src/components/ComputeModal.js

```diff
@@ -195,16 +214,22 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
   })
   // State -- end
 
-  const isPersistentDisk = shouldUsePersistentDisk(sparkMode, currentRuntimeDetails, upgradeDiskSelected)
+  const isPersistentDisk = shouldUsePersistentDisk(runtimeType, currentRuntimeDetails, upgradeDiskSelected)
 
   const isCustomImage = selectedLeoImage === customMode
   const { version, updated, packages, requiresSpark, label: packageLabel } = _.find({ image: selectedLeoImage }, leoImages) || {}
-
-  const minRequiredMemory = sparkMode ? 7.5 : 3.75
+  // The memory sizes below are the minimum required to launch Terra-supported GCP runtimes, based on experimentation.
+  const minRequiredMemory = isDataproc(runtimeType) ? 7.5 : 3.75 // in GB
   const validMachineTypes = _.filter(({ memory }) => memory >= minRequiredMemory, machineTypes)
-  const mainMachineType = _.find({ name: computeConfig.masterMachineType }, validMachineTypes)?.name || getDefaultMachineType(sparkMode)
+  const mainMachineType = _.find({ name: computeConfig.masterMachineType }, validMachineTypes)?.name || getDefaultMachineType(isDataproc(runtimeType))
   const machineTypeConstraints = { inclusion: { within: _.map('name', validMachineTypes), message: 'is not supported' } }
 
+  const isRuntimeRunning = currentRuntimeDetails?.status === 'Running'
+  const shouldDisplaySparkConsoleLink = isDataproc(runtimeType) && currentRuntimeDetails?.runtimeConfig?.componentGatewayEnabled
+  const canManageSparkConsole = shouldDisplaySparkConsoleLink && isRuntimeRunning
+
+  const canUpdateNumberOfWorkers = !currentRuntimeDetails || isRuntimeRunning
+
```

Comment:
The functions right below didn't need to be functions so I converted them to constants. Moved them up so they are in the right section.